src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -388,6 +390,7 @@ public ClusterState execute(final ClusterState currentState) throws Exception {
                                 }
                             }
                         }
+                        mappingValidator.validateMetaDataFields(indexService.mapperService());
```

Comment:
I suspect this does not check much since the mapping update is not applied yet. I guess this is hard since if you put the check after mappings are applied then there is no way to recover from this situation?